llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6322,6 +6333,9 @@ SwitchLookupTable::SwitchLookupTable(
       if (!ConstVal) {
         // This is an undef. We could deal with it, but undefs in lookup tables
         // are very seldom. It's probably not worth the additional complexity.
+        // TODO: In switches with holes and an unreachable default branch, this
```

Comment:
I wasn't planning on fixing it in this PR, since it seemed like too big of a change. I've got another PR lined up which fixes this TODO, which I was planning to open as soon as this gets merged. Would you prefer I made that part of this PR?

Reply:
It's fine with me, but I hope you can restore the unhandled cases to their previous values in this PR.

Reply:
I've done this as well. Is this what you had in mind?

Reply:
I think you can set it to the first value here: https://github.com/llvm/llvm-project/blob/d75f9dd1d29b332bdc51346de63cbc04646354d7/llvm/lib/Transforms/Utils/SimplifyCFG.cpp#L6268. Your test cases reminded me that we can directly provide `undef`/`poison`, so this might not cause significant regressions. Others might have different opinions on this, so I'll request comments from them after you address the `poison` and `undef` mixing issue.